bcrypt.go

```diff
@@ -122,12 +123,20 @@ func HashBytes(password []byte, salt ...[]byte) (hash []byte, err error) {
 	// 			"^\\$2([a-z]?)\\$([0-3][0-9])\\$([\\./A-Za-z0-9]{22}+)"
 
 	// Ok, extract the required information
-	minor := byte('y')
+	minor := byte(0)
 	sr := bytes.NewBuffer(s)
-	if !consume(sr, '$') || !consume(sr, '2') || !consume(sr, 'y') || !consume(sr, '$') {
+
+	if !consume(sr, '$') || !consume(sr, '2') {
 		return nil, InvalidSalt
 	}
 
+	if !consume(sr, '$') {
+		minor, _ = sr.ReadByte()
+		if minor != 'a' || !consume(sr, '$') {
```

Comment:
Rather than reverting it in this PR (since this is kind of a hidden breaking change), I would make a separate PR which reverts this commit, _and_ does something like:
```suggestion
		if (minor != 'a' && minor != 'y') || !consume(sr, '$') {
```
That way, it's not a breaking change, it fixes internally generated salts, _and_ we can pass `$2y$` and then we don't have to modify the returned hash.